src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -0,0 +1,236 @@
+
+        if (disableState) {
+            return;
+        }
+    }
+
+    smButton = currentSMButton;
+    if (smButton == tbLeftAreaSelectManually) {
+        start = sbLeftAreaStart;
+        end = sbLeftAreaEnd;
+    } else {
+        start = sbRightAreaStart;
+        end = sbRightAreaEnd;
+    }
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    auto sequenceSelection = activeSequenceContext->getSequenceSelection();
+    SAFE_POINT(sequenceSelection != nullptr, L10N::nullPointerError("DNASequenceSelection"), );
+
+    updateRegionConnection = connect(sequenceSelection, &DNASequenceSelection::si_onSelectionChanged, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_updateRegion);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_updateRegion() {
+    auto sequenceSelection = qobject_cast<DNASequenceSelection*>(sender());
+    SAFE_POINT(sequenceSelection != nullptr, L10N::nullPointerError("DNASequenceSelection"), );
+
+    const auto& selectedRegions = sequenceSelection->getSelectedRegions();
+    CHECK(selectedRegions.size() != 0, );
+
+    if (selectedRegions.size() > 1) {
+        coreLog.details(tr("Multiple regions selection, the only first is set as primer search area."));
+    }
+    const auto& region = selectedRegions.first();
+    start->setValue(region.startPos + 1);
+    end->setValue(region.endPos());
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_updateMaxValues() {
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    auto seqLength = activeSequenceContext->getSequenceLength();
+    for (const auto& minSb : minMaxValuesSb.keys()) {
+        auto maxSb = minMaxValuesSb[minSb];
+        minSb->setMaximum(maxSb->value() - 1);
+        maxSb->setMinimum(minSb->value() + 1);
+        maxSb->setMaximum(seqLength);
+    }
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_minValueChanged() {
+    auto minSb = qobject_cast<QSpinBox*>(sender());
+    SAFE_POINT(minSb != nullptr, L10N::nullPointerError("QSpinBox"), );
+
+    auto maxSb = minMaxValuesSb.value(minSb);
+    SAFE_POINT(maxSb != nullptr, L10N::nullPointerError("QSpinBox"), );
+
+    maxSb->setMinimum(minSb->value() + 1);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_maxValueChanged() {
+    auto maxSb = qobject_cast<QSpinBox*>(sender());
+    SAFE_POINT(maxSb != nullptr, L10N::nullPointerError("QSpinBox"), );
+
+    auto minSb = minMaxValuesSb.key(maxSb);
+    SAFE_POINT(minSb != nullptr, L10N::nullPointerError("QSpinBox"), );
+
+    minSb->setMaximum(maxSb->value() - 1);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_loadBackbone() {
+    const QString filter = DialogUtils::prepareDocumentsFileFilterByObjType(GObjectTypes::SEQUENCE, true);
```

Comment:
nit: const is not needed: this is a local var with a very limited life range (+ on the next line you do not use const)

Reply:
OK